Reorganize Sudo Section (#122)
moondance-labs/tanssi-docs: builders/manage/sudo/minting.md

---
title: Minting Native Tokens with Sudo 

> description: Learn how to use the Sudo key to manage your appchain and performing the privileged action of minting additional native tokens of your Tanssi appchain.

Reviewer: ```suggestion
description: Learn how to use the Sudo key to manage your appchain and perform the privileged action of minting additional native tokens of your Tanssi appchain.
```

In this guide, you'll learn how to use Sudo to properly mint new native tokens. This comprehensive guide shows how to check the balance of an existing account prior to assigning it a new balance with Sudo access. 

## Checking Prerequisites {: #checking-prerequisites }

For the examples in this guide, you will need to have the following:

 - A Tanssi Appchain (Snap or Dedicated)
 - Your Appchain's Sudo account connected to your Appchain's Polkadot.js Apps. You can refer to the [Managing Sudo guide](/builders/manage/sudo/sudo/#configuring-polkadotjs-apps){target=\_blank} for instructions on injecting your Sudo account into Polkadot.js Apps

If you're unsure what your Tanssi Appchain's Sudo account is, you can find it in your [Tanssi Dashboard](https://apps.tanssi.network/){target=\_blank} underneath the **Properties** section.

> ![Locating your Sudo address on apps.tanssi.network](/images/builders/manage/sudo/minting/minting-1.webp)

Reviewer: the arrow looks a little bit off. it should be inline with the text its pointing to

!!! note
    You should always protect your Sudo account key with the utmost security precautions, as it grants privileged access to your Tanssi Appchain.

## Minting Tokens {: #minting-tokens }

> As you know, the Sudo account has the ability to perform privileged functions, including minting additional tokens. When setting up your Tanssi Appchain on [apps.tanssi.network](https://apps.tanssi.network/){target=\_blank}, you can specify genesis account balances. In other words, you have the ability to endow accounts with initial balances upon launching your Tanssi Appchain. However, you can also mint new tokens after launch with the help of the Sudo account.

Reviewer: ```suggestion
As you know, the Sudo account has the ability to perform privileged functions, including minting additional tokens. When setting up your Tanssi Appchain on the [Tanssi dApp](https://apps.tanssi.network/){target=\_blank}, you can specify genesis account balances. In other words, you have the ability to endow accounts with initial balances upon launching your Tanssi Appchain. However, you can also mint new tokens after launch with the help of the Sudo account.
```

this came back on a recent SEO report, we need to switch to descriptive link text as opposed to writing out the link itself

!!! note
    This tutorial demonstrates assigning arbitrary token balances on a TestNet Appchain that has no value. You should carefully consider the ramifications of creating additional tokens on your own Tanssi Appchain.

### Checking Existing Account Balance {: #checking-existing-account-balance }

The next section will demonstrate how to assign arbitrary token balances to accounts using the Sudo account. This process will overwrite the specified account's existing balance, so verifying the account is empty is a good practice before continuing. To check an account's balance, take the following steps:

> 1. Navigate to the **Developer** tab of [Polkadot.js Apps](#configuring-polkadotjs-apps) and click on **Chain State**

Reviewer: There's a couple of these links, but they need to be updated to point to the page where this section is now

2. Select the **system** pallet to query
3. Select **account**
4. Paste in the account address or select it from the dropdown
5. Press **+** icon
6. You'll see the balance information returned at the bottom, including free, reserved, and frozen balances

![Check balances on Polkadot.js Apps](/images/builders/manage/sudo/minting/minting-2.webp)

### Assigning Balances with Sudo {: #assigning-balances-with-sudo }  

To assign an account balance to an account, make sure to have your Sudo account accessible in [Polkadot.js Apps](#configuring-polkadotjs-apps). Then, take the following steps:

1. Navigate to the **Developer** Tab of Polkadot.js Apps for your Tanssi Appchain
2. Click on **Sudo**. If you do not see **Sudo** in this menu, then you have not associated the Sudo account with Polkadot.js Apps. Make sure that your Sudo account is injected by your wallet and connected to Polkadot.js Apps
3. Select the **balances** pallet
4. Select the **forceSetBalance** method
5. Paste in the account address to endow with tokens or select it from the dropdown

> 6. Enter the amount of tokens to endow the account with. In this example, we specify `9000000000000000000` for nine native tokens. Remember that Tanssi EVM Appchains have 18 decimals while Substrate or Custom Appchains decimals are configured when launching the chain. If you're unsure how many decimals your Tanssi Appchain has, navigate to the **Settings** tab and Click on **Metadata**

Reviewer: ```suggestion
6. Enter the amount of tokens to endow the account with. In this example, we specify `9000000000000000000` for nine native tokens. Remember that Tanssi EVM appchains have 18 decimals, while Substrate or custom appchains configure the decimals when launching the chain. If you're unsure how many decimals your Tanssi appchain has, navigate to the **Settings** tab and click on **Metadata**
```
